reflex/.templates/jinja/web/pages/index.js.jinja2

```diff
@@ -25,7 +25,7 @@ export default function Component() {
 
   // Route after the initial page hydration.
   useEffect(() => {
-    const change_complete = () => Event([E('{{state_name}}.{{const.hydrate}}', {})])
+    const change_complete = () => Event(initialEvents.map((e) => ({...e})))
```

Comment:
Why can't this just be `() => Event(initialEvents)` ?

Reply:
Because with the new client-side routing code, we only update the `router_data` in `applyEvent` if it's not already set. So if we don't spread the `initialEvents` to make a copy, then subsequent calls with hydrate with the wrong router_data.

I don't love it. But we're getting rid of it soon, so this could be considered a workaround until we fully decouple hydrate and `on_load`. (Hydrate doesn't care about the router_data, only the on_load events do).